### ISSUE TITLE
`make compile` fails due to trying to pulling non-existing branch of `apache/camel-quarkus` repo

### ISSUE THREAD

**Issue by chanseokoh** [adoptium/aqa-tests#2738]
**Describe the bug**
`BUILD_LIST=external make compile` fails when checking out `https://github.com/apache/camel-quarkus.git`. The `camel-quarkus` repo doesn't have the `master` branch but `main`. I think they renamed it at some point.

```
     [exec] Step 18/22 : RUN git clone https://github.com/apache/camel-quarkus.git
     [exec]  ---> Running in 443a9032a830
     [exec] Cloning into 'camel-quarkus'...
     [exec] Removing intermediate container 443a9032a830
     [exec]  ---> 063ff5e61422
     [exec] Step 19/22 : WORKDIR /camel-quarkus/
     [exec]  ---> Running in 6aedff57e11b
     [exec] Removing intermediate container 6aedff57e11b
     [exec]  ---> 91cc446a13f1
     [exec] Step 20/22 : RUN git checkout $CAMEL_TAG
     [exec]  ---> Running in a25c978c718f
     [exec] error: pathspec 'master' did not match any file(s) known to git
     [exec] The command '/bin/sh -c git checkout $CAMEL_TAG' returned a non-zero code: 1
     [exec] 

BUILD FAILED
.../aqa-tests/TKG/scripts/build_test.xml:76: The following error occurred while executing this line:
.../aqa-tests/external/build.xml:42: The following error occurred while executing this line:
.../aqa-tests/external/camel/build.xml:25: The following error occurred while executing this line:
.../aqa-tests/external/build.xml:87: exec returned: 1
```

**To Reproduce**
```
export BUILD_LIST=external

git clone https://github.com/adoptium/aqa-tests.git
cd aqa-tests
./get.sh
cd TKG
make compile
```

**Comment by smlambert** [comment]
Note we intend to eventually read from the test.properties file, but for now still read from common_functions.sh, so to fix this issue, updates needed at:

https://github.com/adoptium/aqa-tests/blob/master/external/common_functions.sh#L189 
and 
https://github.com/adoptium/aqa-tests/blob/master/external/camel/test.properties#L5

Also note if you want to run a particular external test, set BUILD_LIST=external/camel or which ever test you want to run in that directory, otherwise we try to compile ALL tests in the external directory, which takes a very long time.

**Comment by unnati914** [comment]
could i work on this?

**Comment by smlambert** [comment]
Yes please @unnati914 !

**Comment by chanseokoh** [comment]
Thanks for the explanation. That helps.

Another error while trying to run all the "external" tests. Is AdoptOpenJDK not being tested daily with "external" tests? I can understand that these tests can be flaky and will probably need constant maintenance, but I am surprised that there are multiple broken tests, meaning that they are probably not being utilized for AdoptOpenJDK 11 at least. Or, is it that running "external" tests are not really worth it?

This time, it's with the `scala` test. It tries to download `https://piccolo.link/sbt-1.2.8.tgz`, which redirects to `https://sbt-downloads.cdnedge.bluemix.net/...`. But the host address is not resolvable.
```
     [echo] Executing external.sh --build --dir scala --tag nightly --version 11 --impl hotspot 
...
     [exec] Step 12/25 : RUN wget --no-check-certificate --no-cookies https://piccolo.link/sbt-${SBT_VERSION}.tgz 	&& wget --no-check-certificate --no-cookies https://piccolo.link/sbt-${SBT_VERSION}.tgz.asc 	&& curl "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | gpg --import 	&& gpg --verify sbt-${SBT_VERSION}.tgz.asc sbt-${SBT_VERSION}.tgz 	&& tar -zvxf sbt-${SBT_VERSION}.tgz -C /opt/ 	&& rm -f sbt-${SBT_VERSION}.tgz 	&& rm -f sbt-${SBT_VERSION}.tgz.asc
     [exec]  ---> Running in 91d663720ab0
     [exec] --2021-07-15 21:08:12--  https://piccolo.link/sbt-1.2.8.tgz
     [exec] Resolving piccolo.link (piccolo.link)... 52.116.127.25, 52.116.122.240
     [exec] Connecting to piccolo.link (piccolo.link)|52.116.127.25|:443... connected.
     [exec] HTTP request sent, awaiting response... 301 Moved Permanently
     [exec] Location: https://sbt-downloads.cdnedge.bluemix.net/releases/v1.2.8/sbt-1.2.8.tgz [following]
     [exec] --2021-07-15 21:08:13--  https://sbt-downloads.cdnedge.bluemix.net/releases/v1.2.8/sbt-1.2.8.tgz
     [exec] Resolving sbt-downloads.cdnedge.bluemix.net (sbt-downloads.cdnedge.bluemix.net)... failed: Name or service not known.
     [exec] wget: unable to resolve host address ‘sbt-downloads.cdnedge.bluemix.net’
```

Another question: I'm still interested in running all "external" tests with `BUILD_LIST=external make compile`, which also tries the `external_custom` test. This particular test is not a self-contained one but is like a hook/template/helper for anyone to run a custom test against an arbitrary repo, right?

**Comment by smlambert** [comment]
we are in the process of overhauling the external tests, so have turned off daily builds... the overhaul is being done under a summer project under the Outreachy program (1 intern working on it).  If you have an interest and are willing to help us lift them up faster, that would be wonderful!  Otherwise, the plan is to continue to work on them for the remainder of the summer term to see if we can get to a sane state again.  

external_custom is the WIP target being worked on under the Outreachy project, please see an overview here: https://github.com/adoptium/aqa-tests/wiki/Outreachy-Simplify-and-Gamify-External-Tests-Project:-Tweetest

**Comment by unnati914** [comment]
I could help if i can :)

**Comment by smlambert** [comment]
Yes please help @unnati914 - I have assigned this issue to you, please create a PR if you have the time and interest.

**Comment by unnati914** [comment]
hi what could configure from this issue is.. that while cloning the repo i have to mention the branch name..  is there anything else i need to do?

**Comment by smlambert** [comment]
All you need to do for this issue is to create a PR with the change to the branch name in the 2 files as described in https://github.com/adoptium/aqa-tests/issues/2738#issuecomment-880274081.

Once you create a PR, the reviewers will run a test to see that the test works.  You can also try to run the test locally following the instructions in https://github.com/adoptium/aqa-tests/tree/master/external/camel#running-external-camel-tests-locally, but that is not a requirement in order to provide a fix.

**Comment by CtripleU** [comment]
Hi @smlambert. I am an Outreachy applicant. Is it okay for me to work on this?

**Comment by sophia-guo** [comment]
Closed as this has been resolved by a couple of updates
